src/model_config_utils.cc

```diff
@@ -1620,6 +1620,32 @@ ValidateModelConfig(
             " cache.");
   }
 
+  // If model_metric is specified, validate it.
+  if (config.has_model_metrics()) {
+#ifdef TRITON_ENABLE_METRICS
+    for (const auto& metric_control : config.model_metrics().metric_control()) {
+      if (metric_control.has_metric_identifier()) {
+        if (metric_control.metric_identifier().family().empty()) {
+          return Status(
+              Status::Code::INVALID_ARG,
+              "metric_identifier must specify 'family'");
+        }
+      } else {
+        return Status(
+            Status::Code::INVALID_ARG,
+            "model_control must specify 'metric_identifier'");
+      }
+
+      if (!metric_control.has_histogram_options()) {
```

Comment:
Shouldn't you check if the histogram is empty?

Reply:
If you are referring to buckets, I don't see a way to check if buckets field is empty or not. The following two protobuf objects are identical examined in GDB. Since buckets `[]` is legal, I think we just have to allow buckets to be omitted.
```
histogram_options: {
    buckets: []
}

histogram_options: {
}
```

Reply:
Discussioned offline. Will disallow users to set empty buckets.